### PR TITLE
Pass selection options to useRealtme hook

### PR DESCRIPTION
**Pull request by kachar**
Related to https://github.com/tmm/react-supabase/discussions/17

With this update the hook `useRealtime` accepts both `string` table name or object `{ table: string } & UseSelectConfig<Data>`

This update is backward compatible so no breaking change is introduced.

Docs are updated with the following example:

```tsx highlight=6
import { useRealtime } from 'react-supabase'

function Page() {
  const [result, reexecute] = useRealtime(
    {
      table: 'todos',
      columns: 'id, username, description',
      filter: (query) => query.eq('completed', false),
    },
    (data, payload) => data.username === payload.username,
  )

  return ...
}
```

### CODE REVIEW

**Comment by kachar** [comment]
@tmm Any idea why the Size PR check is failing ? It doesn't seem related to the code updates

**Comment by tmm** [comment]
@kachar can you grab latest `main`? should fix that problem.

File: src/hooks/realtime/use-realtime.ts [hunk 1]
```diff
@@ -31,17 +31,22 @@ export type UseRealtimeCompareFn<Data = any> = (
 type CompareFnDefaultData<Data> = Data & { id: any }
 
 export function useRealtime<Data = any>(
-    table: string,
+    options: string | ({ table: string } & UseSelectConfig<Data>),
     compareFn: UseRealtimeCompareFn<Data> = (a, b) =>
         (<CompareFnDefaultData<Data>>a).id ===
         (<CompareFnDefaultData<Data>>b).id,
 ): UseRealtimeResponse<Data> {
```
Comment (tmm) [inline src/hooks/realtime/use-realtime.ts:38]: Can we keep the `table` as a string and add a new `config` parameter?

```
export type UseRealtimeConfig<Data> = {
    select?: Omit<UseSelectConfig<Data>, 'pause'>
}

export function useRealtime<Data = any>(
    table: string,
    config: UseRealtimeConfig<Data>,
    compareFn: UseRealtimeCompareFn<Data> = (a, b) =>
        (<CompareFnDefaultData<Data>>a).id ===
        (<CompareFnDefaultData<Data>>b).id,
): UseRealtimeResponse<Data>
```

I appreciate wanting to keep things backward compatible, but it's early enough in the project that I'm fine introducing a breaking change for the sake of a better API foundation for the future.

**Review by tmm — CHANGES_REQUESTED** [review]

File: src/hooks/realtime/use-realtime.ts (hunk 1 above)
Comment (kachar) [reply src/hooks/realtime/use-realtime.ts]: @tmm Code is updated

Indeed this update makes the arguments and the fn api much more explicit and open for extension :rocket:

**Pull request merged**